### PR TITLE
PAAS-186 do not show projects that we would never select / do not have roles

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/paas

### CODE REVIEW

File: plugins/kubernetes/app/views/admin/kubernetes/deploy_group_roles/_form.html.erb [hunk 1]
```diff
@@ -12,7 +12,8 @@
     <div class="form-group">
       <%= form.label :project_id, 'Project', class: "col-lg-2 control-label" %>
       <div class="col-lg-2">
-        <%= form.select :project_id, current_user.administrated_projects.pluck(:name, :id), {}, class: "form-control", disabled: @deploy_group_role.persisted? %>
+        <% projects = current_user.administrated_projects.where(id: Kubernetes::Role.pluck(:project_id)).pluck(:name, :id) %>
```
Comment (jonmoter) [inline plugins/kubernetes/app/views/admin/kubernetes/deploy_group_roles/_form.html.erb]: How does it work to do a class-level `pluck`?

Comment (grosser) [inline plugins/kubernetes/app/views/admin/kubernetes/deploy_group_roles/_form.html.erb]: class level pluck means all projects that have any roles configured ... this endpoint is not scoped by project, so needs to be global

Comment (jonmoter) [inline plugins/kubernetes/app/views/admin/kubernetes/deploy_group_roles/_form.html.erb]: How about creating a scope or class-level method on Project, so you can say `Project.with_kubernetes_roles` or something like that?

Comment (grosser) [inline plugins/kubernetes/app/views/admin/kubernetes/deploy_group_roles/_form.html.erb]: done

**Comment by jonmoter** [comment]
👍 on 💚

**Pull request merged**